### PR TITLE
Add NEST Desktop target

### PR DESCRIPTION
**Pull request by pnbabu**
This PR adds NEST Desktop as a target platform to NESTML. The output of the code generator, in this case, will be a `.json` file containing the details of the neuron model file(s). These json files will be used by the NEST Desktop GUI to select models from a model repository and use them for further simulations.

Note: A part of the documentation for adding new targets is borrowed from #729

### CODE REVIEW

**Comment by jougs** [comment]
@jens-br

**Comment by babsey** [comment]
I apologize for the long silence to react to this PR.

Currently, I got somehow installation/compilation errors of NESTML on my machine (Ubuntu 20.04, Python 3.8).
Thus, I will upgrade to Ubuntu 22.04 in next month and will report then when I am able to test it.

File: pynestml/codegeneration/resources_nest_desktop/@NEURON_NAME@.json.jinja2 [hunk 1]
```diff
@@ -0,0 +1,17 @@
+{
+    "id": "{{neuronName}}",
+    "elementType": "neuron",
+    "label": "{{neuronName}}}}",
```
Comment (babsey) [inline pynestml/codegeneration/resources_nest_desktop/@NEURON_NAME@.json.jinja2:4]:
```suggestion
    "label": "{{neuronName}}",
```

**Review by babsey — APPROVED** [review]
LGTM! Only one suggestion.

I tested it in a conda environment with Python 3.9.
In IPython I ran the commands:
```
from pynestml.frontend.pynestml_frontend import generate_target
generate_target(
    input_path="/<path>/<to>/nestml/models/neurons/izhikevich.nestml", 
    logging_level="INFO", 
    target_path="/tmp/nestml-component", 
    target_platform="NEST_DESKTOP"
)
```

**Comment by babsey** [comment]
Friendly ping to prevent stale status.

File: pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2 [hunk 2]
```diff
@@ -0,0 +1,10 @@
+    {%- set name = variable.get_symbol_name() %}
+    {%- set type_symbol = variable.get_type_symbol().print_symbol() %}
+    {%- set default_value = parameters[variable.get_symbol_name()] %}
+      {
+        "id": "{{name}}",
+        "label": "{{name}}",
+        "unit": "{{type_symbol}}",
+        "value": "{{default_value}}",
+        "input": "arrayInput"
```
Comment (babsey) [inline pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2:9]: Can you please remove this line `input`? It is not valid.

File: pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2 [hunk 3]
```diff
@@ -0,0 +1,10 @@
+    {%- set name = variable.get_symbol_name() %}
+    {%- set type_symbol = variable.get_type_symbol().print_symbol() %}
+    {%- set default_value = parameters[variable.get_symbol_name()] %}
+      {
+        "id": "{{name}}",
+        "label": "{{name}}",
```
Comment (babsey) [inline pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2:6]: Is it possible to take label from the comment part?

**Review by babsey — CHANGES_REQUESTED** [review]
It works. I ran pytest on this testing file. The JSON output looks fine.

I added some minor requests to change.

File: pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2 (hunk 2 above)
Comment (pnbabu) [reply pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2]: Removed.

File: pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2 (hunk 3 above)
Comment (pnbabu) [reply pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2]: Done.

File: pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2 [hunk 4]
```diff
@@ -0,0 +1,9 @@
+    {%- set name = variable.get_symbol_name() %}
+    {%- set type_symbol = variable.get_type_symbol().print_symbol() %}
+    {%- set default_value = parameters[variable.get_symbol_name()] %}
+      {
+        "id": "{{name}}",
+        "label": "{{variable.print_comment()}}",
```
Comment (babsey) [inline pynestml/codegeneration/resources_nest_desktop/ParameterDeclaration.jinja2:6]:
```suggestion
        "label": "{{variable.print_comment().strip()}}",
```

**Review by babsey — APPROVED** [review]
Thank your for updating! 

LGTM except 2 inline suggestions.

**Comment by babsey** [comment]
LGTM! 

@clinssen ?

File: .github/workflows/nestml-build.yml [hunk 5]
```diff
@@ -123,7 +123,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        nest_branch: ["v2.20.2", "v3.0", "v3.7", "master"]
```
Comment (clinssen) [inline .github/workflows/nestml-build.yml:126]: Should the v3.8_rc1 changes be in this branch/PR?

File: tests/nest_desktop_tests/nest_desktop_code_generator_test.py [hunk 6]
```diff
@@ -0,0 +1,64 @@
+# -*- coding: utf-8 -*-
```
Comment (clinssen) [inline tests/nest_desktop_tests/nest_desktop_code_generator_test.py:1]: The test is currently not being run in CI. Could you add a separate stage for it, just like we have for the Python code generator and the compartmental code generator?

@babsey: is there any integration test we could do with NEST Desktop in CI?

File: doc/running/index.rst [hunk 7]
```diff
@@ -101,7 +101,7 @@ The following default values are used, corresponding to the command line default
      - *no default*
    * - target_platform
      - str
-     - "NEST"
```
Comment (clinssen) [inline doc/running/index.rst:104]: this table lists default values, so this change should be undone

File: doc/running/index.rst [hunk 8]
```diff
@@ -186,7 +186,7 @@ This will generate, compile, build, and install the code for a set of specified
    * - ``--target_path``
      - (Optional) Path to target directory where generated code will be written into. Default is ``target``, which will be created in the current working directory if it does not yet exist.
    * - ``--target_platform``
-     - (Optional) The name of the target platform to generate code for. Default is ``NEST``.
+     - (Optional) The name of the target platform to generate code for. The available targets are ``NEST`` and ``NEST_DESKTOP``. Default is ``NEST``.
```
Comment (clinssen) [inline doc/running/index.rst:189]: Python and compartmental and SpiNNaker seem to be missing from the list. Could we just link back to the paragraph "Supported target platforms" (https://nestml.readthedocs.io/en/latest/running/index.html#supported-target-platforms)?

File: doc/running/index.rst [hunk 9]
```diff
@@ -201,3 +201,18 @@ This will generate, compile, build, and install the code for a set of specified
      - (Optional) Enable development mode: code generation is attempted even for models that contain errors, and extra information is rendered in the generated code. Default is OFF.
    * - ``--codegen_opts``
      - (Optional) Path to a JSON file containing additional options for the target platform code generator. A list of available options can be found under the section "Code generation options" for your intended target platform on the page :ref:`Running NESTML`.
+
+NEST Desktop target
```
Comment (clinssen) [inline doc/running/index.rst:205]: Could you instead add a button like for Python-standalone and SpiNNaker under "Supported target platforms" (https://nestml.readthedocs.io/en/latest/running/index.html#supported-target-platforms)?

**Review by clinssen — CHANGES_REQUESTED** [review]

File: tests/nest_desktop_tests/nest_desktop_code_generator_test.py (hunk 6 above)
Comment (babsey) [reply tests/nest_desktop_tests/nest_desktop_code_generator_test.py:1]: Unfortunately, I do not have any kind of testing suites of NEST Desktop in CI.

File: .github/workflows/nestml-build.yml (hunk 5 above)
Comment (pnbabu) [reply .github/workflows/nestml-build.yml]: Merged it by mistake. Removed it now.

**Comment by babsey** [comment]
Friendly ping.  Is it possible to use it in v8.0?

**Review by clinssen — APPROVED** [review]
Thanks very much for the PR!

**Pull request merged**